docs/en/table_design/table_types/primary_key_table.md

```diff
@@ -1,159 +1,212 @@
+
+<Tabs groupId="primary key Index">
+
+  <TabItem value="example1" label="Persistent primary keyIndex" default>
+
+When `enable_persistent_index` is set to `true` (default), the primary key indexes can be persisted to the disk. During loading, a small portion of the primary key indexes is loaded in memory, while the majority is stored on disk to avoid taking up too much memory. In general, query and update performance of the table with the persistent primary key indexes is nearly equivalent to that of the table with the fully in-memory primary key indexes.
+
+If the disk is a SSD, it is recommended to set it to true. If the disk is HDD and the load frequency is not high, you can also set it to true.
+
+:::note
 
-## Usage notes
+Since v3.1.4, Primary Key tables created in StarRocks shared-data clusters further support index persistence onto local disks.
 
-- Take note of the following points about the primary key of a table:
-  - The primary key is defined by using the `PRIMARY KEY` keyword.
+:::
 
-  - The primary key must be created on columns on which unique constraints are enforced, and the names of the primary key columns cannot be changed.
+</TabItem>
 
-  - The primary key columns can be any of the following data types: BOOLEAN, TINYINT, SMALLINT, INT, BIGINT, LARGEINT, STRING, VARCHAR, DATE, and DATETIME. However, the primary key columns cannot be defined as `NULL`.
+<TabItem value="example2" label="fully in-memory primary key index">
 
-  - The partition column and the bucket column must participate in the primary key.
+When `enable_persistent_index` is set to `false`, the primary key indexes are not persisted to the disk,  that is, the primary key indexes are fully stored in memory. During loading, the primary key indexes of tablets related to the data loaded will be loaded into memory, which may result in higher memory consumption. (If a tablet has not had data loaded for a long time, its primary key index will be released from memory).
 
-  - The number and total length of primary key columns must be properly designed to save memory. We recommend that you identify columns whose data types occupy less memory and define those columns as the primary key. Such data types include INT and BIGINT. We recommend that you do not let a column of the VARCHAR data type to participate in the primary key.
+When using the fully in-memory primary key indexes, it is recommended to follow the following guidelines when designing the Primary Key to control memory usage of the primary key indexes:
```

Comment:
![image](https://github.com/StarRocks/starrocks/assets/57167462/c4a187d4-a0d1-4ca4-ab62-85fa46564fba)
-> you are advised to